config/aws-config.js

```diff
@@ -18,13 +18,29 @@ const localstack = process.env.LOCALSTACK || 0
 
 const intermediateRole = process.env.AWS_INTERMEDIATE_ROLE_ARN || 0
 
+const stsEndpoint = process.env.AWS_STS_ENDPOINT || 0
+const ssmEndpoint = process.env.AWS_SSM_ENDPOINT || 0
+const smEndpoint = process.env.AWS_SM_ENDPOINT || 0
+
 let secretsManagerConfig = {}
 let systemManagerConfig = {}
 let stsConfig = {
   region: process.env.AWS_REGION || 'us-west-2',
   stsRegionalEndpoints: process.env.AWS_STS_ENDPOINT_TYPE || 'regional'
 }
 
+if (smEndpoint) {
+  secretsManagerConfig.endpoint = smEndpoint
+}
+
+if (ssmEndpoint) {
+  systemManagerConfig.endpoint = ssmEndpoint
+}
+
+if (stsEndpoint) {
+  stsConfig.endpoint = stsEndpoint
+}
+
 if (localstack) {
```

Comment:
I'd like to consolidate the configuration for e2e tests and your custom endpoints using `AWS_*_ENDPOINT`. This way we can hit two birds with one stone:
1. have end-to-end tests for your feature
2. simplify the configuration code

I think it makes sense to use your `AWS_*_ENDPOINT` in the [e2e test code](https://github.com/external-secrets/kubernetes-external-secrets/search?q=LOCALSTACK_SM_URL) and remove the localstack/e2e-specific confinguration variables.

What do you think?

Reply:
@moolen - that totally makes sense.  Is it possible to merge this PR and then consolidate the e2e test endpoints in a separate PR?  This PR makes the endpoints configurable and the next PR can change the e2e tests to leverage this configuration.
It will be cleaner this way as there will be some questions as to how and where to best make the e2e changes, which would be easier to review as part of a separate change.

Reply:
> consolidate the e2e test endpoints in a separate PR?

Yes, for sure! 
I just noticed the `AWS_*_ENDPOINT` aren't yet added to the helm chart. Could you please add/document them there aswell?

Reply:
I added the other env vars to the readme and commented in the default values.yaml